syntax = "proto3";

package bazi.fortune;

// Bazi Fortune Service - 大运流年计算服务

// 请求消息
message BaziFortuneRequest {
  string solar_date = 1;      // 阳历日期，格式 YYYY-MM-DD
  string solar_time = 2;      // 阳历时间，格式 HH:MM
  string gender = 3;          // 性别，male/female 或 男/女
  string current_time = 4;     // 当前时间，ISO 8601 格式（可选）
}

// 响应消息（使用 JSON 字符串存储复杂结构）
message BaziFortuneResponse {
  string detail_json = 1;      // 详细运势信息（JSON 字符串）
  string metadata_json = 2;   // 元数据（JSON 字符串）
}

// 健康检查请求
message HealthCheckRequest {}

// 健康检查响应
message HealthCheckResponse {
  string status = 1;
}

// Bazi Fortune 服务定义
service BaziFortuneService {
  // 计算大运流年
  rpc CalculateDayunLiunian(BaziFortuneRequest) returns (BaziFortuneResponse);
  
  // 健康检查
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

